pkg/migrations/migration_version.go

```diff
@@ -104,5 +103,36 @@ func (m *MigrationVersion) atomicWrite() error {
 	if err != nil {
 		return err
 	}
-	return ioutils.AtomicWriteFile(filepath.Join(m.dbPath, MigrationVersionFile), bytes, migrationVersionFileMode)
+	return atomicWriteFile(filepath.Join(m.dbPath, MigrationVersionFile), bytes, migrationVersionFileMode)
+}
+
+func atomicWriteFile(filename string, bytes []byte, mode os.FileMode) error {
+	tempFile, err := os.CreateTemp(filepath.Split(filename))
+	if err != nil {
+		return err
+	}
+	tempName := tempFile.Name()
+
+	if _, err := tempFile.Write(bytes); err != nil {
+		_ = tempFile.Close()
+		return errors.Wrapf(err, "could not write to %q", tempName)
+	}
+
+	if err := tempFile.Close(); err != nil {
+		return errors.Wrapf(err, "could not close %q", tempName)
+	}
+
+	if err := os.Chmod(tempName, mode); err != nil {
+		return errors.Wrapf(err, "could not chmod %q", tempName)
+	}
+
+	if _, err := os.Stat(tempName); err != nil {
```

Comment:
Is this needed?

Reply:
not too important for this backport, but this may not be necessary, and may potentially be removed in `master`

Reply:
I agree but I decided to implement it and this will be removed with removal of rocksdb